tests/bdd/tokens/test_generate.py

```diff
@@ -0,0 +1,169 @@
+# SPDX-FileCopyrightText: 2022-2023 VMware Inc
+#
+# SPDX-License-Identifier: MIT
+import ast
+
+from pytest_bdd import given, scenario, then, when
+from pytest_bdd.parsers import parse
+
+
+@scenario(
+    "../features/tokens/generate.feature",
+    "Admin uses HTTP API to generate a token",
+)
+def test_admin_uses_http_api_to_generate_a_token():
+    """Admin uses HTTP API to generate a token."""
+
+
+@given("the admin has the admin password", target_fixture="admin_passwd")
+def the_admin_has_the_admin_password(get_admin_pwd):
+    return get_admin_pwd
+
+
+@given(
+    "the admin gets an 'access_token' by logging in to '/api/v1/token' with a "
+    "'write:token' scope",
+    target_fixture="access_token",
+)
+def the_admin_has_generated_an_access_token_with_write_token(access_token):
+    return access_token
+
+
+@given(
+    parse("the admin adds Authorization Bearer {token} in the 'headers'"),
+    target_fixture="headers",
+)
+def the_admin_adds_authorization_token_in_the_headers(access_token, token):
```

Comment:
I cannot find where is the `token` fixture nor do I see it as a variable in the BDD scenario.
Where is implemented?

Reply:
Hm, I thought the description should exactly match the one described in the text, but it seems that you can assign a value to it that way.